sdk/storage/storage-queue/src/QueueClient.ts

```diff
@@ -1103,22 +1102,20 @@ export class QueueClient extends StorageClient {
       //  URL may look like the following
       // "https://myaccount.queue.core.windows.net/myqueue?sasString".
       // "https://myaccount.queue.core.windows.net/myqueue".
-      // or an emulator URL that starts with the endpoint `http://127.0.0.1:10001/devstoreaccount1`
+      // IP style Endpoints - `http://127.0.0.1:10001/devstoreaccount1`
+      // http://localhost:10001/devstoreaccount1/queuename
 
-      let urlWithoutSAS = this.url.split("?")[0]; // removing the sas part of url if present
-      urlWithoutSAS = urlWithoutSAS.endsWith("/") ? urlWithoutSAS.slice(0, -1) : urlWithoutSAS; // Slicing off '/' at the end if exists
+      const parsedUrl = URLBuilder.parse(this.url);
 
-      // http://127.0.0.1:10001/devstoreaccount1
-      const emulatorQueueEndpoint = getValueInConnString(
-        DevelopmentConnectionString,
-        "QueueEndpoint"
-      );
-
-      if (this.url.startsWith(emulatorQueueEndpoint)) {
-        // Emulator URL starts with `http://127.0.0.1:10001/devstoreaccount1`
-        queueName = urlWithoutSAS.match(emulatorQueueEndpoint + "/([^/]*)")![1];
+      if (parsedUrl.getHost()!.split(".")[1] === "queue") {
```

Comment:
Very slight worry: since storage accounts can have arbitrary vanity URLs, and those URLs might contain a "queue" subdomain, is this kind of URL inference a good idea?

Reply:
I'm assuming here that the endpoint only looks like   `${defaultEndpointsProtocol}://${accountName}.queue.${endpointSuffix};`
with an `accountname` having no [DOT]s in it.

Thoughts, @XiaoningLiu ?

Reply:
Storage account name must be between 3 and 24 characters in length and use numbers and lower-case letters only. No dot is allowed in account name.

Reply:
@XiaoningLiu is the endpoint always a url like @HarshaNalluru has above, or can it be a custom vanity URL?

Reply:
@bterlson Currently, the storage endpoint in different regions are like what @HarshaNalluru mentions. Only, `endpointSuffix` is different in different regions.

Reply:
@XiaoningLiu I was asking about custom domains for queues but I see it's not a thing, so no worries here!